framework/include/executioners/Executioner.h

```diff
@@ -117,19 +117,13 @@ class Executioner : public MooseObject,
   virtual bool lastSolveConverged() const = 0;
 
   /// Return the underlining FEProblemSolve object.
-  FEProblemSolve & feProblemSolve() { return _feproblem_solve; }
+  virtual const FEProblemSolve * feProblemSolve() const { return nullptr; }
 
   /// Return underlying PicardSolve object.
-  PicardSolve & picardSolve()
-  {
-    mooseDeprecated("picardSolve() is deprecated. Use FixedPointSolve() instead.");
-    if (_iteration_method == "picard")
-      return *(dynamic_cast<PicardSolve *>(_fixed_point_solve.get()));
-    else
-      mooseError("Cannot return a PicardSolve if the iteration method is not Picard.");
-  }
+  virtual const PicardSolve * picardSolve() const { return nullptr; }
 
-  FixedPointSolve & fixedPointSolve() { return *_fixed_point_solve; }
+  virtual const FixedPointSolve * fixedPointSolve() const { return nullptr; }
+  virtual FixedPointSolve * fixedPointSolve() { return nullptr; }
```

Comment:
I really don't like this. Where we were previously assured of always having objects, now we are not. We do not even have any error checking if somehow a derived class doesn't override this and then someone calls `operator->` and gets a seg-fault. Basically I'm not supportive of any design where the default in the base is to error, and in this case it's even worse than erroring, it's seg-faulting.